### PR TITLE
Add retain cdf option for median calculations when computing info fields

### PR DESCRIPTION
**Pull request by klaricch**

### CODE REVIEW

File: gnomad/utils/sparse_mt.py [hunk 1]
```diff
@@ -394,6 +412,9 @@ def get_as_info_expr(
         of alternate alleles e.g., `hl.range(1, hl.len(mt.alleles))`
     :param treat_fields_as_allele_specific: Treat info fields as allele-specific.
         Defaults to False.
+    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs) as an annotation
+        for median_agg_fields. Keeping the CDFs is useful for annotations that require calculating the median
```
Comment (jkgoodrich) [inline gnomad/utils/sparse_mt.py:416]:
```suggestion
        for `median_agg_fields`. Keeping the CDFs is useful for annotations that require calculating the median
```

File: gnomad/utils/sparse_mt.py [hunk 2]
```diff
@@ -505,6 +528,9 @@ def get_site_info_expr(
     :param sum_agg_fields: Fields to aggregate using sum.
     :param int32_sum_agg_fields: Fields to aggregate using sum using int32.
     :param median_agg_fields: Fields to aggregate using (approximate) median.
+    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs) as an annotation
+        for median_agg_fields. Keeping the CDFs is useful for annotations that require calculating the median
```
Comment (jkgoodrich) [inline gnomad/utils/sparse_mt.py:532]:
```suggestion
        for `median_agg_fields`. Keeping the CDFs is useful for annotations that require calculating the median
```

File: gnomad/utils/sparse_mt.py [hunk 3]
```diff
@@ -577,6 +605,9 @@ def default_compute_info(
         for VQSR.
     :param ac_filter_groups: Optional dictionary of sample filter expressions to compute
         additional groupings of ACs. Default is None.
+    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs) as an annotation
+        for median_agg_fields. Keeping the CDFs is useful for annotations that require calculating the median
```
Comment (jkgoodrich) [inline gnomad/utils/sparse_mt.py:609]:
```suggestion
        for `median_agg_fields`. Keeping the CDFs is useful for annotations that require calculating the median
```

**Review by jkgoodrich — COMMENTED** [review]
A couple questions and some small changes

File: gnomad/utils/sparse_mt.py [hunk 4]
```diff
@@ -253,6 +257,17 @@ def _agg_list_to_dict(
...
 
+    if retain_cdfs:
+        cdf_median_agg_fields = {}
+        # Store values for each median agg fields in a new dictionary with "_cdf"
+        # appended to the annotation name.
+        for k, v in median_agg_fields.items():
+            cdf_median_agg_fields[f"{k}_cdf"] = v
+        # Append the cdf annotations to the aggs list.
+        aggs.append(
+            (cdf_median_agg_fields, lambda x: hl.agg.approx_cdf(x, k=200, _raw=True))
```
[the remark this answers is not in the document]
Comment (klaricch) [reply gnomad/utils/sparse_mt.py:268]: was a recommended default to start with here:
https://hail.zulipchat.com/#narrow/stream/123010-Hail-Query-0.2E2-support/topic/Median.20of.20Histogram/near/395262880

File: gnomad/utils/sparse_mt.py [hunk 5]
```diff
@@ -195,6 +197,12 @@ def _get_info_agg_expr(
         array.
     :param prefix: Optional prefix for the fields. Used for adding 'AS_' in the AS case.
     :param treat_fields_as_allele_specific: Treat info fields as allele-specific. Defaults to False.
+    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs) as an annotation
+        for `median_agg_fields`. Keeping the CDFs is useful for annotations that require calculating the median
+        across combined datasets at a later stage. Default is False.
+    :param cdf_k: Parameter controlling the accuracy vs. memory usage tradeoff when retaining CDFs. A higher
+        value of `cdf_k` results in a more accurate CDF approximation but increases memory usage and computation
+        time. Default is 200.
```
Comment (jkgoodrich) [inline gnomad/utils/sparse_mt.py:205]: Sorry, please excuse the OCD

```suggestion
    :param treat_fields_as_allele_specific: Treat info fields as allele-specific. 
        Defaults to False.
    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs) 
        as an annotation for `median_agg_fields`. Keeping the CDFs is useful for 
        annotations that require calculating the median across combined datasets at a 
        later stage. Default is False.
    :param cdf_k: Parameter controlling the accuracy vs. memory usage tradeoff when 
        retaining CDFs. A higher value of `cdf_k` results in a more accurate CDF 
        approximation but increases memory usage and computation time. Default is 200.
```

File: gnomad/utils/sparse_mt.py [hunk 6]
```diff
@@ -394,6 +421,12 @@ def get_as_info_expr(
         of alternate alleles e.g., `hl.range(1, hl.len(mt.alleles))`
     :param treat_fields_as_allele_specific: Treat info fields as allele-specific.
         Defaults to False.
+    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs) as an annotation
+        for `median_agg_fields`. Keeping the CDFs is useful for annotations that require calculating the median
+        across combined datasets at a later stage. Default is False.
+    :param cdf_k: Parameter controlling the accuracy vs. memory usage tradeoff when retaining CDFs. A higher
+        value of `cdf_k` results in a more accurate CDF approximation but increases memory usage and computation
+        time. Default is 200.
```
Comment (jkgoodrich) [inline gnomad/utils/sparse_mt.py:429]:
```suggestion
    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs)
        as an annotation for `median_agg_fields`. Keeping the CDFs is useful for
        annotations that require calculating the median across combined datasets at a
        later stage. Default is False.
    :param cdf_k: Parameter controlling the accuracy vs. memory usage tradeoff when
        retaining CDFs. A higher value of `cdf_k` results in a more accurate CDF
        approximation but increases memory usage and computation time. Default is 200.
```

File: gnomad/utils/sparse_mt.py [hunk 7]
```diff
@@ -505,6 +542,12 @@ def get_site_info_expr(
     :param sum_agg_fields: Fields to aggregate using sum.
     :param int32_sum_agg_fields: Fields to aggregate using sum using int32.
     :param median_agg_fields: Fields to aggregate using (approximate) median.
+    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs) as an annotation
+        for `median_agg_fields`. Keeping the CDFs is useful for annotations that require calculating the median
+        across combined datasets at a later stage. Default is False.
+    :param cdf_k: Parameter controlling the accuracy vs. memory usage tradeoff when retaining CDFs. A higher
+        value of `cdf_k` results in a more accurate CDF approximation but increases memory usage and computation
+        time. Default is 200.
```
Comment (jkgoodrich) [inline gnomad/utils/sparse_mt.py:550]:
```suggestion
    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs) 
        as an annotation for `median_agg_fields`. Keeping the CDFs is useful for 
        annotations that require calculating the median across combined datasets at a 
        later stage. Default is False.
    :param cdf_k: Parameter controlling the accuracy vs. memory usage tradeoff when 
        retaining CDFs. A higher value of `cdf_k` results in a more accurate CDF 
        approximation but increases memory usage and computation time. Default is 200.
```

File: gnomad/utils/sparse_mt.py [hunk 8]
```diff
@@ -577,6 +624,12 @@ def default_compute_info(
         for VQSR.
     :param ac_filter_groups: Optional dictionary of sample filter expressions to compute
         additional groupings of ACs. Default is None.
+    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs) as an annotation
+        for `median_agg_fields`. Keeping the CDFs is useful for annotations that require calculating the median
+        across combined datasets at a later stage. Default is False.
+    :param cdf_k: Parameter controlling the accuracy vs. memory usage tradeoff when retaining CDFs. A higher
+        value of `cdf_k` results in a more accurate CDF approximation but increases memory usage and computation
+        time. Default is 200.
```
Comment (jkgoodrich) [inline gnomad/utils/sparse_mt.py:632]:
```suggestion
    :param retain_cdfs: If True, retains the cumulative distribution functions (CDFs) 
        as an annotation for `median_agg_fields`. Keeping the CDFs is useful for 
        annotations that require calculating the median across combined datasets at a 
        later stage. Default is False.
    :param cdf_k: Parameter controlling the accuracy vs. memory usage tradeoff when 
        retaining CDFs. A higher value of `cdf_k` results in a more accurate CDF 
        approximation but increases memory usage and computation time. Default is 200.
```

**Review by jkgoodrich — APPROVED** [review]
Just some docstring param formatting, sorry about the OCD

Otherwise, LGTM!

**Pull request merged**